### ISSUE TITLE
React: Masthead - screen reader is reading  controls which are not available visually.

### ISSUE THREAD

**Issue by Praveen-sr** [carbon-design-system/carbon-for-ibm-dotcom#4199]
## Detailed description
 
On Voiceover iPhone
Issue: 
1. visually only Search  edit combo , search icon and  close icon is present.
 but for screen reader users screen reader is reading controls which are not available visually.
for example hamburger menu, IBM logo link,  user profile button etc.
2. after entering the search term visually the options are appearing but   screen reader users are not able to navigate through the options within the combo box

Expected:
1. screen reader should not read visually hidden components
2. the user should be able to navigate through the options  using left and right swipe and make the selection.
